File: R/Day3/day3mathandstats.R
data <- datasets::data
# mean 
base::mean(data$mpg)

# trimmed_mean
base::mean(data$mpg, trim = 0.2)

# weighted mean
stats::weighted.mean(data$mpg, data$cyl)

# median
stats::median(data$mpg)

# weighted median
install.packages('spatstat')
library(spatstat)
spatstat.geom::weighted.median(data$mpg, data$cyl)

# percentile
stats::quantile(data$mpg)

# 25th 50th and 95h percentile
stats::quantile(data$mpg, c(.25,.5,.95))

# since 50th percentile is equal to median following shoud be TRUE
stats::quantile(data$mpg, c(.5)) == stats::median(data$mpg)

File: Python/Day3/day3mathandstats.py
import pandas as pd
data = pd.read_csv('/content/mtcars.csv')
# mean
print('Mean: ',data['mpg'].mean())

# trimmed mean from scipy
import scipy.stats as sc
print('Trimmed Mean: ',sc.trim_mean(data['mpg'], 0.1))

# median
print('Median', data['mpg'].median())

import numpy as np
print('Weighted Mean: ',np.average(data['mpg'], weights = data['cyl']))

#pip install wquantiles
import wquantiles
# weighted median
print('Weighted Median: ',wquantiles.median(data['mpg'], weights = data['cyl']))